lib/vector-common/src/event_test_util.rs

```diff
@@ -5,9 +5,32 @@ thread_local! {
     static EVENTS_RECORDED: RefCell<HashSet<String>> = RefCell::new(HashSet::new());
 }
 
-#[must_use]
-pub fn contains_name(name: &str) -> bool {
-    EVENTS_RECORDED.with(|events| events.borrow().iter().any(|event| event.ends_with(name)))
+/// Returns Ok(()) if:
+///   - `allow_double_emission` is true and any event contains `name`.
```

Comment:
When would we want to allow for double emission?

Reply:
In the case of the happy path `assert_source_compliance` , there were a lot of cases that did emit one or more of the specified events multiple times ... I took that to mean it is possible.
But if not, then I'll need to change this to not allow any double emissions, and go through and fix all the places that do double emit - or make that a TODO for another time.

Reply:
I wonder if, instead of `allow_double_emission`, we could have the caller specify the number of times we expect the event to be emitted?

Reply:
Yeah, I'm struggling to think of why double emission would be good. I figured it was there as an escape hatch of "broken, fix later"

Reply:
Yeah, you guys know better than me on this one...
I'll change it to not allow double emission at all and see if I can collect the places that it fails due to that, and maybe I can share that and get input on whether each falls under the category of either needs to be changed to only emit once, or can be allowed to emit N times ?